src/Refinement/RefinedTriangulations.jl

```diff
@@ -0,0 +1,126 @@
+
+
+struct RefinedTriangulation{Dc,Dp,A<:Triangulation{Dc,Dp},B<:RefinedDiscreteModel} <: Triangulation{Dc,Dp}
+  trian::A
+  model::B
+end
+
+function get_refined_model(t::RefinedTriangulation)
+  return t.model
+end
+
+# Wrap Triangulation API
+function Geometry.get_background_model(t::RefinedTriangulation)
+  get_background_model(t.trian)
+end
+
+function Geometry.get_grid(t::RefinedTriangulation)
+  get_grid(t.trian)
+end
+
+function Geometry.get_glue(t::RefinedTriangulation,::Val{d}) where d
+  get_glue(t.trian,Val(d))
+end
+
+function Base.view(t::RefinedTriangulation,ids::AbstractArray)
+  v = view(t.trian,ids)
+  return RefinedTriangulation(v,t.model)
+end
+
+# Wrap constructors for RefinedDiscreteModel
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,filter::AbstractArray) where d
+  
+  trian = Triangulation(ReferenceFE{d},get_model(model),filter)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(
+  ::Type{ReferenceFE{d}},model::RefinedDiscreteModel,labels::FaceLabeling;kwargs...) where d
+  trian = Triangulation(ReferenceFE{d},get_model(model),labels;kwargs...)
+  return RefinedTriangulation(trian,model)
+end
+
+function Geometry.Triangulation(trian::RefinedTriangulation,args...;kwargs...)
+  return RefinedTriangulation(Triangulation(trian.trian),trian.model)
+end
+
+# Domain changes
+# TODO: This assumes we have the same type of triangulation on both refinement levels!
+#       we might want to change this in the future when doing hybrid methods etc...
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::RefinedTriangulation)
+  (strian === ttrian) && (return true)
+  smodel = get_refined_model(strian)
+  tmodel = get_refined_model(ttrian)
+  a = get_parent(tmodel) === smodel # tmodel = refine(smodel)
+  b = get_parent(smodel) === tmodel # smodel = refine(tmodel)
+  return a || b
+end
+
+function Geometry.is_change_possible(strian::RefinedTriangulation,ttrian::T) where {T <: Triangulation}
```

Comment:
Note that (perhaps you already are aware) that using `ttrian::T,  T <: Triangulation` is equivalent to `ttrian::Triangulation`

Reply:
The same comment applies below for many other functions

Reply:
Resolved in [0f1ed7a](https://github.com/gridap/Gridap.jl/pull/838/commits/0f1ed7a8307cc19f07baddb70e6d499ca8ddecf2)